app/scripts/modules/core/src/securityGroup/AllSecurityGroupsCtrl.js

```diff
@@ -53,15 +50,14 @@ module.exports = angular.module('spinnaker.core.securityGroup.all.controller', [
       $scope.$evalAsync(() => {
         securityGroupFilterService.updateSecurityGroups(app);
         groupsUpdated();
-        // Timeout because the updateSecurityGroups method is debounced by 25ms
-        $timeout(() => { this.initialized = true; }, 50);
       });
     };
 
     let groupsUpdated = () => {
       $scope.$applyAsync(() => {
         $scope.groups = securityGroupFilterModel.groups;
         $scope.tags = securityGroupFilterModel.tags;
+        this.initialized = this.initialized || app.securityGroups.loaded;
```

Comment:
this was one of the main problems: we were always setting the state to initialized, even if the application's security groups were not yet loaded.

Reply:
good jorb